File: solutions/rust/0060.rs
impl Solution {
    /*
     * 排列序列 / Permutation Sequence
     *
     * 返回由数字 1..n 组成的所有排列中，按字典序排序后的第 k 个排列（1-indexed）。
     *
     * 时间复杂度: O(n^2)
     *   - 每一位选择时需要从数组中移除一个元素，remove 是 O(n)
     * 空间复杂度: O(n)
     *   - 需要存储阶乘数组和候选数字
     */
    pub fn get_permutation(n: i32, k: i32) -> String {
        if n <= 0 || k <= 0 {
            return String::new();
        }

        let n_usize = n as usize;

        // 计算阶乘数组 fact[i] = i!
        let mut fact: Vec<i64> = vec![1; n_usize + 1];
        for i in 1..=n_usize {
            fact[i] = fact[i - 1] * i as i64;
        }

        if (k as i64) > fact[n_usize] {
            return String::new();
        }

        // 候选数字 1..n
        let mut nums: Vec<i32> = (1..=n).collect();

        // 将 k 转为 0-index
        let mut k0: i64 = (k - 1) as i64;
        let mut result = String::with_capacity(n_usize);

        for i in (1..=n_usize).rev() {
            let f = fact[i - 1];
            let idx = (k0 / f) as usize;
            let val = nums.remove(idx);
            result.push_str(&val.to_string());
            k0 %= f;
        }

        result
    }

    /// 中文包装方法，保持与其它语言版本一致
    pub fn 排列序列(n: i32, k: i32) -> String {
        Self::get_permutation(n, k)
    }
}

File: solutions/typescript/0060.ts
/**
 * 排列序列 - 返回由 1..n 组成的所有排列中字典序第 k 个（1-indexed）
 *
 * 时间复杂度: O(n²)，每次从数组中 remove 为 O(n)
 * 空间复杂度: O(n)
 */
function getPermutation(n: number, k: number): string {
    if (n <= 0 || k <= 0) return "";

    const fact: number[] = [1];
    for (let i = 1; i <= n; i++) fact[i] = fact[i - 1] * i;
    if (k > fact[n]) return "";

    const nums: number[] = [];
    for (let i = 1; i <= n; i++) nums.push(i);

    let k0 = k - 1; // 转为 0-index
    const res: string[] = [];

    for (let i = n; i >= 1; i--) {
        const f = fact[i - 1];
        const idx = Math.floor(k0 / f);
        res.push(String(nums[idx]));
        nums.splice(idx, 1);
        k0 %= f;
    }

    return res.join("");
}

function 排列序列(...args: any[]): any {
    const n = args[0] as number;
    const k = args[1] as number;
    return getPermutation(n, k);
}

export default 排列序列;
